src/screens/Podcast/Podcast.tsx

```diff
@@ -1,112 +1,151 @@
+import { useDispatch } from 'react-redux'
+import { useRoute, useNavigation } from '@react-navigation/native'
+import {
+  View,
+  Image,
+  ImageBackground,
+  ActivityIndicator,
+  TouchableOpacity
+} from 'react-native'
+import { useAppSelector } from 'src/hooks'
+import { DEFAULT_IMAGE_BASE64 } from 'src/common/constants/defaultImage'
+import {
+  DEFAULT_EPISODES_PAGINATION,
+  DEFAULT_EPISODES_LIMIT
+} from './common/constants'
+import { DataStatus, NavigationScreen } from 'src/common/enums'
+import { PodcastRouteProp, PodcastNavigationProp } from './common/types'
+import {
+  loadPodcast as loadPodcastAction,
+  loadEpisodesByPodcastId as loadEpisodesByPodcastIdAction
+} from 'src/store/actions'
 import { Heading, HeadingType, PlainText } from 'src/components'
-import { EpisodeBlock } from './components'
+import { EpisodeList, NoPodcast } from './components'
 import BackButton from 'src/assets/images/backButton.svg'
 import CircleIcon from 'src/assets/images/circle.svg'
 import styles from './styles'
 
-type Episode = {
-  id: number
-  name: string
-  createdAt: string
-}
+const Podcast: React.FC = () => {
+  const { podcast, episodes, dataStatus, totalCount, hasMoreEpisodes } =
+    useAppSelector(({ podcast }) => ({
+      podcast: podcast.podcast,
+      episodes: podcast.episodes,
+      dataStatus: podcast.dataStatus,
+      totalCount: podcast.totalCount,
+      hasMoreEpisodes: podcast.hasMoreEpisodes
+    }))
 
-type Podcast = {
-  image: string
-  background: string
-  name: string
-  author: string
-  description: string
-  episodes: Episode[]
-}
+  const route = useRoute<PodcastRouteProp>()
+  const navigation = useNavigation<PodcastNavigationProp>()
+
+  const isLoading = dataStatus === DataStatus.PENDING
 
-const mockedPodacst: Podcast = {
-  background:
-    'http://res.cloudinary.com/hmqu8gtpn/image/upload/v1629474209/2/wqbm1x3q0h6ea2ebrnze.jpg',
-  image:
-    'http://res.cloudinary.com/hmqu8gtpn/image/upload/v1629474124/2/ttp0xygzbdyrn0yyxfnk.jpg',
-  name: 'Cartoon Soundtracks',
-  author: 'Eugenius',
-  description: "Cartoon's Soundtracks Disney, Dreamworks and others",
-  episodes: [
-    {
-      id: 1,
-      name: 'Kingdom Dance',
-      createdAt: 'August 16, 2021'
-    },
-    {
-      id: 2,
-      name: 'You are Welcome',
-      createdAt: 'August 20, 2021'
-    },
-    {
-      id: 3,
-      name: 'This is Berk',
-      createdAt: 'August 26, 2021'
+  const dispatch = useDispatch()
+
+  const fetchEpisodes = (pagination = DEFAULT_EPISODES_PAGINATION) => {
+    dispatch(
+      loadEpisodesByPodcastIdAction({
+        podcastId: Number(route.params.id),
+        filter: pagination
+      })
+    )
+  }
+
+  const handleLoadEpisodes = () => {
+    if (hasMoreEpisodes) {
+      fetchEpisodes({
+        offset: episodes.length,
+        limit: DEFAULT_EPISODES_LIMIT
+      })
     }
-  ]
-}
+  }
+
+  useEffect(() => {
+    dispatch(loadPodcastAction(Number(route.params.id)))
+    handleLoadEpisodes()
```

Comment:
Overall looks good 👍🏻

I recommend you to clear state after unmounting the component. It will help to avoid some UI overlapping when you choosing another podcast.
```suggestion
    handleLoadEpisodes()
    return () => {
        dispatch(resetState())
    }
```

Reply:
I added resetState action in next commit. Thank you!